### PR TITLE
Feature/observable key groups collection tests

### PR DESCRIPTION
**Pull request by nkrisko**
### Description

Unit Tests for ObservableKeyGroupsCollection

### API Changes

 None

### Platforms Affected

- Core (all platforms)

### Behavioral/Visual Changes

None

### Before/After Screenshots

Not applicable

### PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Softeq/XToolkit.WhiteLabel/blob/master/.github/CONTRIBUTING.md) document
- [x] My code follows the [code styles](https://github.com/Softeq/dotnet-guidelines)
- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Softeq.XToolkit.Common/Collections/ObservableKeyGroupsCollection.cs [hunk 1]
```diff
@@ -2,450 +2,705 @@
...
-        public void UnionSortedGroups(
-            ObservableKeyGroupsCollection<TKey, TValue> newCollection,
-            IEqualityComparer<TValue> itemComparer)
+        private class Group : List<TValue>, IGrouping<TKey, TValue>
         {
-            var newKeys = Keys.Union(newCollection.Keys).OrderBy(x => x).ToArray();
+            public TKey Key { get; }
 
-            for (var i = 0; i < newKeys.Length; i++)
+            public Group(KeyValuePair<TKey, IList<TValue>> keyValuePair)
```
Comment (wcoder) [inline Softeq.XToolkit.Common/Collections/ObservableKeyGroupsCollection.cs:699]: Stiles: properties under the ctor

**Review by wcoder — CHANGES_REQUESTED** [review]
Looks good, but doesn't work with Android/iOS Playground samples (grouped list/collection).

Also, I put some comments.

File: Softeq.XToolkit.Common/Collections/ObservableKeyGroupsCollection.cs [hunk 2]
```diff
@@ -2,450 +2,705 @@
...
+                {
+                    (_items.IndexOf(item),
+                        new NotifyGroupCollectionChangedEventArgs<TValue>(NotifyCollectionChangedAction.Reset, default, default))
+                };
+
+            OnChanged(
+                default,
+                default,
+                default,
+                clearGroupEvent); // check, how check key ?
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Common/Collections/ObservableKeyGroupsCollection.cs:249]: yep

File: Softeq.XToolkit.Common/Collections/ObservableKeyGroupsCollection.cs [hunk 3]
```diff
@@ -2,450 +2,705 @@
...
+            var insertedGroups = InsertGroupsWithoutNotify(index, items, _withoutEmptyGroups);
+            if (insertedGroups == null)
+            {
+                return;
+            }
 
-        public void AddRangeToGroups(IList<TValue> listItem, Func<TValue, TKey> selector = null)
-        {
-            AddRangeToGroups(listItem, x => x, selector);
+            var newItems = new Collection<(int, IReadOnlyList<TKey>)>
```
[the remark this answers is not in the document]
Comment (nkrisko) [reply Softeq.XToolkit.Common/Collections/ObservableKeyGroupsCollection.cs:108]: It was before me

**Review by wcoder — APPROVED** [review]

**Pull request merged**